crankshaft-console/src/view/resources.rs

```diff
@@ -0,0 +1,62 @@
+//! Renders the resources view.
+use ratatui::Frame;
+use ratatui::layout::Constraint;
+use ratatui::layout::Direction;
+use ratatui::layout::Layout;
+use ratatui::style::Color;
+use ratatui::style::Modifier;
+use ratatui::style::Style;
+use ratatui::widgets::Block;
+use ratatui::widgets::Borders;
+use ratatui::widgets::Cell;
+use ratatui::widgets::Row;
+use ratatui::widgets::Table;
+
+use crate::state::resource::ResourceState;
+
+/// Renders the resources view.
+pub(crate) fn render_resource(frame: &mut Frame<'_>, resource_state: &ResourceState) {
+    let area = Layout::default()
+        .direction(Direction::Vertical)
+        .margin(1)
+        .constraints([Constraint::Min(0)])
+        .split(frame.size());
+
+    let headers = ["Task ID", "CPU", "Max CPU", "Memory", "Max Memory", "Nodes"]
+        .iter()
+        .map(|h| {
+            Cell::from(*h).style(
+                Style::default()
+                    .fg(Color::Yellow)
+                    .add_modifier(Modifier::BOLD),
+            )
+        });
+    let header = Row::new(headers).height(1).bottom_margin(1);
+
+    let r = resource_state.resources();
+    let row = vec![
```

Comment:
Looking at the `ratatui` documentation, I think we can probably get away with using static arrays here for both `row` and `rows` rather than allocating a `Vec`.

It's worth a try, at least.